### PR TITLE
Adds the Blueshield from Oracle

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

ports https://github.com/OracleStation/OracleStation/pull/3

This adds the Blueshield Job to the game.

## Why It's Good For The Game

A good amount of people wanted this, and i think it's a good idea too, so why not, more jobs are always a good thing.

### What is the Blueshields Job?

The Blueshield is a Special Type of Role that has the job of protecting the Heads of Staff of the station, **THEY ARE NOT SECURITY!!!!!!!!** They have their own office on every map, access to Command Comms, Bridge Access, Crew monitor console, and a Special Energy Stun Revoler (Shoots Disabler/Lethal Lasers projectiles)

image

Boxstation Blueshield Office, located next to the gravity generator (Gravity Generator was slightly moved to the left)
Metastation Blueshield Office Located next to the Gateway (Replaced the Gateway Equipment Room)
Kilostation Blueshield Office, located on the bridge (Replaced the Council Room)
Deltastation Blueshield Office (Same as Metastation)
Pubbystation Blueshield Office (Replaced the Teleporter Room, moved the Teleporter room to below Tool storage, i didn't see any other way to add it to this map)

Also i had to create some insane piping to add disposals to some maps, sorry to any people that will see that.

## Changelog
:cl:
add: Added a new Job, the Blueshield, Their job is to Protect the Heads of Staff, they are not Security.
/:cl:

### CODE REVIEW

**Comment by TeomanTheGreat** [comment]
Yes

File: code/__DEFINES/jobs.dm [hunk 1]
```diff
@@ -40,6 +39,7 @@
 #define CARGOTECH		(1<<7)
 #define MINER			(1<<8)
 #define LAWYER			(1<<9)
+#define BLUESHIELD (1<<14)
 #define CHAPLAIN		(1<<10)
 #define CLOWN			(1<<11)
 #define MIME			(1<<12)
```
Comment (PowerfulBacon) [inline code/__DEFINES/jobs.dm]: Blueshield has the same thing as gimmick

File: code/modules/jobs/job_types/blueshield.dm [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  outfit = /datum/outfit/job/blueshield
+
+  access = list(ACCESS_SECURITY, ACCESS_SEC_DOORS, ACCESS_COURT, ACCESS_WEAPONS,
+			            ACCESS_MEDICAL, ACCESS_ENGINE, ACCESS_CHANGE_IDS, ACCESS_AI_UPLOAD, ACCESS_EVA, ACCESS_HEADS,
+			            ACCESS_ALL_PERSONAL_LOCKERS, ACCESS_MAINT_TUNNELS, ACCESS_BAR, ACCESS_JANITOR, ACCESS_CONSTRUCTION, ACCESS_MORGUE,
+			            ACCESS_CREMATORIUM, ACCESS_KITCHEN, ACCESS_CARGO, ACCESS_MAILSORTING, ACCESS_QM, ACCESS_HYDROPONICS, ACCESS_LAWYER,
+			            ACCESS_THEATRE, ACCESS_CHAPEL_OFFICE, ACCESS_LIBRARY, ACCESS_RESEARCH, ACCESS_MINING, ACCESS_MINING_STATION,
+			            ACCESS_HOP, ACCESS_RC_ANNOUNCE, ACCESS_KEYCARD_AUTH, ACCESS_GATEWAY, ACCESS_MINERAL_STOREROOM, ACCESS_BLUESHIELD)
+  minimal_access = list(ACCESS_FORENSICS_LOCKERS, ACCESS_SEC_DOORS, ACCESS_MEDICAL, ACCESS_CONSTRUCTION, ACCESS_ENGINE, ACCESS_MAINT_TUNNELS,
+                  ACCESS_RESEARCH, ACCESS_RC_ANNOUNCE, ACCESS_KEYCARD_AUTH, ACCESS_HEADS, ACCESS_BLUESHIELD, ACCESS_WEAPONS)
```
Comment (PowerfulBacon) [inline code/modules/jobs/job_types/blueshield.dm]: This is a significant amount more access on lowpop

File: code/modules/projectiles/guns/energy/energy_gun.dm [hunk 3]
```diff
@@ -66,6 +66,15 @@
 	ammo_x_offset = 4
 	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF
 
+/obj/item/gun/energy/e_gun/blueshield
+	name = "advanced stun revolver"
+	desc = "An advanced stun revolver with the capacity to shoot both electrodes and lasers."
+	icon_state = "bsgun"
+	item_state = "gun"
+	force = 7
+	ammo_type = list(/obj/item/ammo_casing/energy/electrode/hos, /obj/item/ammo_casing/energy/laser/hos)
```
Comment (PowerfulBacon) [inline code/modules/projectiles/guns/energy/energy_gun.dm]: I don't like the idea of something that shoots lasers or tasers, tasers kind of are terribly unfun when it comes to combat and are used on HOS gun only for a reason

**Review by PowerfulBacon — COMMENTED** [review]
This should just be a random maint role imo, rather than a proper full out one

**Comment by PowerfulBacon** [comment]
Also, putting THEY ARE NOT SECURITY!! then giving them a taser revolver and security gear isn't going to stop them from validhunting antags instead of protecting the heads which is a pretty boring job

**Comment by PowerfulBacon** [comment]
Also this is significantly outdated really in concept, since it _was_ their 3rd PR out of 700, probably why it still uses tasers

**Comment by ike709** [comment]
This is a terrible, terrible idea. By far the worst role ever added to Oracle.

It is quite literally Validhunter: The Job.

**Comment by ghost** [comment]
> This is a terrible, terrible idea. By far the worst role ever added to Oracle.
> 
> It is quite literally Validhunter: The Job.

that's already security though.

**Comment by ike709** [comment]
That's not a good excuse to add another validhunter role.

**Comment by AnjelicaChaykovsky** [comment]
new valid hunter job  ? Me like

**Comment by ghost** [comment]
> new valid hunter job ? Me like

no

**Comment by VictorVow** [comment]
After seeing this on Wasp I've concluded that bodyguarding all Heads as a single individual is not possible, and can only be done by a full Security team or a dedicated Bodyguard role. A Bodyguard random gimmick role (which starts mindshielded) could be assigned a random Head to guard and this would fulfil most of this PRs objective.

**Comment by eeSPee** [comment]
Remove the sec gear, but give them the ability to call EngineeringRT, JanitorRT, nook codes or classic ERT once per round?

Keep the name Blueshield, give it a HoP mini egun and an office.

**Comment by ghost** [comment]
> Remove the sec gear, but give them the ability to call EngineeringRT, JanitorRT, nook codes or classic ERT once per round?
> 
> Keep the name Blueshield, give it a HoP mini egun and an office.

to be honest that sounds more like a NT representative job or internal affairs.

**Comment by ghost** [comment]
regarding the point that some people would play this as a validhunter, role, that can be made a administrative issue, like it's on paradise or the role could be just whitelisted.

**Comment by ike709** [comment]
> Adding a job that exists specifically to validhunt then expecting admins to deal with the validhunters

lol

**Comment by ghost** [comment]
> > Adding a job that exists specifically to validhunt then expecting admins to deal with the validhunters
> 
> lol

thats not their job.

**Comment by PowerfulBacon** [comment]
> 
> 
> > > Adding a job that exists specifically to validhunt then expecting admins to deal with the validhunters
> > 
> > 
> > lol
> 
> thats not their job.

Players tend to not use things the way they are intended if there is a better alternative

**Comment by ghost** [comment]
> > > > Adding a job that exists specifically to validhunt then expecting admins to deal with the validhunters
> > > 
> > > 
> > > lol
> > 
> > 
> > thats not their job.
> 
> Players tend to not use things the way they are intended if there is a better alternative



> regarding the point that some people would play this as a validhunter, role, that can be made a administrative issue, like it's on paradise or the role could be just whitelisted.

**Comment by ghost** [comment]
why is mapdiffs failing

**Comment by ike709** [comment]
> or the role could be just whitelisted.

We're not implementing role whitelisting for one (terrible) role.

**Comment by ghost** [comment]
> regarding the point that some people would play this as a validhunter, role, that can be made a administrative issue,

**Comment by Cenrus** [comment]
naw. We have enough jobs as is

**Pull request closed without merging**